src/metabase/api/cache.clj

```diff
@@ -68,98 +62,40 @@
                                     "question"  :model/Card)
                                   :id [:in ids]))))
 
-(defn- audit-caching-change! [id prev new]
-  (events/publish-event!
-   :event/cache-config-update
-   {:user-id  api/*current-user-id*
-    :model    :model/CacheConfig
-    :model-id id
-    :details  {:model     (or (:model prev) (:model new))
-               :model-id  (or (:model_id prev) (:model_id new))
-               :old-value (dissoc prev :model :model_id)
-               :new-value (dissoc new :model :model_id)}}))
-
 (api/defendpoint GET "/"
   "Return cache configuration."
   [:as {{:strs [model collection]
          :or   {model "root"}}
         :query-params}]
-  {model      (ms/QueryVectorOf CachingModel)
+  {model      (ms/QueryVectorOf cache-config/CachingModel)
    ;; note that `nil` in `collection` means all configurations not scoped to any particular collection
    collection [:maybe ms/PositiveInt]}
   (validation/check-has-application-permission :setting)
-  (let [items (if (premium-features/enable-cache-granular-controls?)
-                (t2/select :model/CacheConfig
-                           :model [:in model]
-                           {:left-join [:report_card      [:and
-                                                           [:= :model [:inline "question"]]
-                                                           [:= :model_id :report_card.id]
-                                                           [:= :report_card.collection_id collection]]
-                                        :report_dashboard [:and
-                                                           [:= :model [:inline "dashboard"]]
-                                                           [:= :model_id :report_dashboard.id]
-                                                           [:= :report_dashboard.collection_id collection]]]
-                            :where     [:case
-                                        [:= :model [:inline "question"]]  [:!= :report_card.id nil]
-                                        [:= :model [:inline "dashboard"]] [:!= :report_dashboard.id nil]
-                                        :else                             true]})
-                (t2/select :model/CacheConfig :model "root"))]
-    {:data (mapv cache-config/row->config items)}))
+  (let [models (if (premium-features/enable-cache-granular-controls?)
+                 model
+                 ["root"])]
```

Comment:
why don't we throw an error in this case tho?

Reply:
Hmmm good question, I was trying to be more liberal in what we accept when it's possible to. Like if we can return something meaningful and it won't hurt us, why not?

Reply:
Well, I'm not sure. IMO, specifying a param is intentional, so you'd expect it to return something different than the default behavior. This could be unexpected, though.

Reply:
Yeah, makes sense, let it be error. Done in last change.